src/values/AbstractObjectValue.js

```diff
@@ -233,9 +245,29 @@ export default class AbstractObjectValue extends AbstractValue {
       invariant(ob2 instanceof ObjectValue);
       let p1 = ob1.$GetPrototypeOf();
       let p2 = ob2.$GetPrototypeOf();
-      let joinedObject = Join.joinValuesAsConditional(this.$Realm, cond, p1, p2);
+      let joinedObject = Join.joinValuesAsConditional(realm, cond, p1, p2);
       invariant(joinedObject instanceof AbstractObjectValue);
       return joinedObject;
+    } else if (this.kind === "explicit conversion to object") {
+      let primitiveValue = this.args[0];
+      switch (primitiveValue.getType()) {
+        case BooleanValue:
+          return realm.intrinsics.BooleanPrototype;
+        case IntegralValue:
+        case NumberValue:
+          return realm.intrinsics.NumberPrototype;
+        case StringValue:
+          return realm.intrinsics.StringPrototype;
+        case SymbolValue:
+          return realm.intrinsics.SymbolPrototype;
+        default:
+          let result = AbstractValue.createFromBuildFunction(realm, ObjectValue, [primitiveValue], ([p]) => {
+            invariant(realm.preludeGenerator !== undefined);
+            return t.callExpression(realm.preludeGenerator.memoizeReference("Object.getPrototypeOf"), [p]);
```

Comment:
JSC doesn't support Object.getPrototypeOf, and __proto__ should be used instead. See here: https://github.com/facebook/prepack/blob/5a7c47c41614401eaac71bd2754fd29785df1c68/src/serializer/ResidualHeapSerializer.js#L305 Some general helper function would be appropriate.